mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -335,6 +335,9 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
       throw new SparkException(msg)
     }
 
+    require(numFeatures > 0 || $(fitIntercept),
+      "Specified model is empty with neither intercept nor feature.")
```

Comment:
The message is a bit cryptic.  How about "GeneralizedLinearRegression was given data with 0 features, and with Param fitIntercept set to false.  To fit a model with 0 features, fitIntercept must be set to true."

Other than this, your PR looks good to me.  Thanks!